vm/src/obj/objsequence.rs

```diff
@@ -57,7 +57,7 @@ pub trait PySliceableSequence {
     }
 }
 
-impl PySliceableSequence for Vec<PyObjectRef> {
+impl<T: Clone> PySliceableSequence for Vec<T> {
```

Comment:
To my opinion, this makes the code somewhat too generic. When will this be used without PyObjectRef? Do we have other uses of types which implement clone which must be sliced?